File: Calorie_counting/elves.js
const {readFileSync } = require('fs');
const { performance } = require('perf_hooks');

var startTime = performance.now()
function syncReadFile(filename) {
  
  const data = readFileSync(filename, 'utf-8');

  const arr = data.split(/\n\n/);

  let result = []

  arr.map((elf, index) => {
    result.push(0)
    elf.split(/\n/).map((cal) => {
      result[index] += parseInt(cal)
    })
  })
  console.log(Math.max(...result))

  let sortedList = result.sort(function(a,b){return b - a}).slice(0 ,3)

  const topSum = sortedList.reduce((topSum, current) => topSum + current);

  console.log(topSum)

  return arr;
}
var endTime = performance.now()

console.log(`${endTime - startTime} milliseconds`)

syncReadFile('./numbers.txt');

File: Calorie_counting/elves.py
def main():
    result_list = []
    with open('numbers.txt', 'r') as file:
        data = file.read()
    for index, elf in enumerate(data.split('\n\n')):
        result_list.append(0)
        for cal in elf.split('\n'):
            result_list[index] += int(cal)
    print(max(result_list))

    most_caloric = []
    for i in range(3):
        greater_cal = max(result_list)
        most_caloric.append(greater_cal)
        result_list.remove(greater_cal)
    print(sum(most_caloric))

if __name__ == '__main__':
    main()

    